taxonomy-manager-engine/src/main/java/com/digirati/taxman/analysis/IdWithPosition.java

```diff
@@ -0,0 +1,25 @@
+package com.digirati.taxman.analysis;
+
+public class IdWithPosition<IdT> {
```

Comment:
Let's give this a better class name. Something that identifies it as a "match". We should also store a reference to the matched tokens instead of just the begin/end positions.